fe/fe-core/src/main/java/com/starrocks/qe/scheduler/DefaultWorkerProvider.java

```diff
@@ -326,8 +339,23 @@ private static ImmutableMap<Long, ComputeNode> buildComputeNodeInfo(SystemInfoSe
                 }
                 computeNodes.put(computeNode.getId(), computeNode);
             }
-            return ImmutableMap.copyOf(computeNodes);
+        } else { //numUsedComputeNodes >= idToComputeNode.size()
```

Comment:
This branch and the previous branch `else if (numUsedComputeNodes < idToComputeNode.size()) {` should be merged to one, and follow the same steps:
1. Get enough compute nodes from `idToComputeNode`, until `computeNodes.size() < numUsedComputeNodes`.
2. If the policy is `ALL_NODES`, then continually get enough compute nodes from `idToBackend`, until `computeNodes.size() < numUsedComputeNodes`.

Otherwise, the result `computeNodes` may contain some non-available compute nodes, but `computeNodes.size() == numUsedComputeNodes`.